plugins/modules/win_scoop.ps1

```diff
@@ -0,0 +1,234 @@
+      $res = Run-Command -Command "powershell.exe -" -stdin $install_script -environment $environment
+      if ($res.rc -ne 0) {
+        $module.Result.rc = $res.rc
+        $module.Result.stdout = $res.stdout
+        $module.Result.stderr = $res.stderr
+        $module.FailJson("Scoop bootstrap installation failed.")
+      }
+      $module.Warn("Scoop was missing from this system, so it was installed during this task run.")
+    }
+    $module.Result.changed = $true
+
+    # Refresh PATH
+    $env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path", "User")
+
+    # locate the newly installed scoop.ps1
+    $scoop_app = Get-Command -Name scoop.ps1 -Type ExternalScript -ErrorAction SilentlyContinue
+  }
+
+  if ($module.CheckMode -and $null -eq $scoop_app) {
+    $module.Result.skipped = $true
+    $module.Result.msg = "Skipped check mode run on win_scoop as scoop.ps1 cannot be found on the system"
+    $module.ExitJson()
+  }
+
+  if ($null -eq $scoop_app -or -not (Test-Path -LiteralPath $scoop_app.Path)) {
+    $module.FailJson("Failed to find scoop.ps1, make sure it is added to the PATH")
+  }
+
+  return $scoop_app.Path
+}
+
+function Get-ScoopPackages {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path
+  )
+
+  $command = Argv-ToString -arguments @("powershell.exe", $scoop_path, "export")
+  $res = Run-Command -Command $command
+  if ($res.rc -ne 0) {
+    $module.Result.command = $command
+    $module.Result.rc = $res.rc
+    $module.Result.stdout = $res.stdout
+    $module.Result.stderr = $res.stderr
+    $module.FailJson("Error checking installed packages")
+  }
+
+  $res.stdout -split "`n" |
+  Select-String '(.*?) \(v:(.*?)\) \[(.*?)\]' |
+  ForEach-Object {
+    [PSCustomObject]@{
+      Package = $_.Matches[0].Groups[1].Value
+      Version = $_.Matches[0].Groups[2].Value
+      Bucket  = $_.Matches[0].Groups[3].Value
+    }
+  }
+}
+
+function Get-InstallScoopPackageArguments {
+  $arguments = [System.Collections.Generic.List[String]]@()
+
+  if ($architecture) {
+    $arguments.Add("--arch")
+    $arguments.Add($architecture)
+  }
+  if ($global) {
+    $arguments.Add("--global")
+  }
+  if ($independent) {
+    $arguments.Add("--independent")
+  }
+  if ($no_cache) {
+    $arguments.Add("--no-cache")
+  }
+  if ($skip_checksum) {
+    $arguments.Add("--skip")
+  }
+
+  return , $arguments
+}
+
+function Install-ScoopPackage {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path,
+    [Parameter(Mandatory = $true)] [String[]]$packages
+  )
+  $arguments = [System.Collections.Generic.List[String]]@("powershell.exe", $scoop_path, "install")
+  $arguments.AddRange($packages)
+
+  $common_args = Get-InstallScoopPackageArguments
+  $arguments.AddRange($common_args)
+
+  $command = Argv-ToString -arguments $arguments
+
+  if (-not $module.CheckMode) {
+    $res = Run-Command -Command $command
+    $module.Result.rc = $res.rc
```

Comment:
So I see we aren't checking the return code, is there any error validation that we can do to make sure the install worked?

Reply:
I copied the return code check from `Get-ScoopPackages` to `Install-ScoopPackage` and `Uninstall-ScoopPackage`.